Automattic/wp-calypso
# Domains: Update Domain Management header

#### Changes proposed in this Pull Request
This PR updates the domain management header section according to the new design.
This is part of the domain management redesign project described in pcYYhz-m2-p2.

#### Preview
##### Old header

<img width="300" src="https://user-images.githubusercontent.com/18705930/145070882-a51c176e-c809-4d26-bd80-d0a651e062a8.png"/>

<img width="300" src="https://user-images.githubusercontent.com/18705930/145070914-7158dc9d-1bf9-42e9-8f0e-1150186228b0.png"/>

##### New header

<img width="300" src="https://user-images.githubusercontent.com/18705930/145266490-3a8d279c-3042-4a2e-a160-c8078055391f.png"/>

<img width="300" src="https://user-images.githubusercontent.com/18705930/145266337-1aadf2c2-4e6c-4cb0-919d-e657ebf27a03.png"/>

<img width="300" src="https://user-images.githubusercontent.com/18705930/145106607-5ce2307f-669c-4cd6-8f6f-7bc95824ee0f.png"/>


#### Testing instructions
- Build this branch locally or open the live Calypso link;
- Go to "Upgrades > Domains";
- Select one of your registered domains or domain connections;
- If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag;
- Please check that:
  - The newly redesigned header is present and correct;
  - The badges are shown according to the domain status:
    - Type:
      - Registered domain: no differentiation badges;
      - Transferred domains: `Domain Transfer` badge;
      - Connected domains: `Registered with an external provider` badge;
    - All the other logic badge logic - which one to show, whether it's a success or a warning badge - will be the same used in the domain's statuses in the domains list (`resolve-domain-status.js`);
  - Mobile view looks correct as shown in the screenshot above;
  - Design looks the same as in hlVh2q24ad6MCwlwNnE9PQ-fi-600%3A52892.
